scripts/push_sdk.sh

```diff
@@ -1,94 +1,144 @@
 #!/usr/bin/env bash
 set -ex
 
-SCRIPT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+LANGUAGE=$1
 
-if [[ $GITHUB_RUN_NUMBER != "" ]]; then
-    BUILD_NUMBER=$GITHUB_RUN_NUMBER
-else
-    echo "Usage: $0 BUILD_NUMBER"
-    exit 1
+ORGANIZATION_NAME="criteo"
+REPOSITORY_NAME="criteo-api-${LANGUAGE}-sdk"
+
+GENERATOR_REPO_DIR=$GITHUB_WORKSPACE
+SDK_REPO_DIR=$RUNNER_TEMP
+
+TAG_VERSION="v$GITHUB_RUN_NUMBER"
+
+if [ "$LANGUAGE" = "" ]; then
+  echo "[ERROR] LANGUAGE not set"
+  exit 1
+fi
+
+if [ "$GENERATOR_REPO_DIR" = "" ]; then
+  echo "[ERROR] GENERATOR_REPO_DIR not set"
+  exit 1
+fi
+
+if [ "$SDK_REPO_DIR" = "" ]; then
+  echo "[ERROR] SDK_REPO_DIR not set"
+  exit 1
+fi
+
+if [ "$GITHUB_ACTOR" = "" ]; then
+  echo "[ERROR] GITHUB_ACTOR not set"
+  exit 1
+fi
+
+if [ "$SDK_REPO_PRIVATE_KEY" = "" ]; then
+  echo "[ERROR] SDK_REPO_PRIVATE_KEY not set"
+  exit 1
+fi
+
+if [[ $TAG_VERSION == "" ]]; then
+  echo "[ERROR] TAG_VERSION is not defined"
+  exit 1
 fi
 
+setup_ssh() {
+  eval "$(ssh-agent -s)"
+  ssh-add - <<< "${SDK_REPO_PRIVATE_KEY}"
+}
+
 git_clone() {
-  git clone --depth 1 https://${GH_TOKEN}@github.com/$1.git
+  echo "[INFO] Cloning $ORGANIZATION_NAME/$REPOSITORY_NAME repository..."
+
+  cd $SDK_REPO_DIR
+  git clone --depth 1 git@github.com:$ORGANIZATION_NAME/$REPOSITORY_NAME.git
+  SDK_REPO_DIR="$SDK_REPO_DIR/$REPOSITORY_NAME"
+
+  echo "[INFO] Success. Repository cloned at $SDK_REPO_DIR"
+}
+
+remove_previous_sdks() {
+  echo "[INFO] Removing previous SDKs..."
+
+  sdks_directory="$SDK_REPO_DIR/sdks/$LANGUAGE"
+
+  if [[ -d $sdks_directory ]]; then
+    cd $sdks_directory
+    rm -rf *
+    echo "[INFO] Success."
+  else
+    echo "[WARN] Directory $REPO_NAME/sdks doesn't exists, skipping."
+  fi
+}
+
+copy_new_sdks() {
+  echo "[INFO] Copying new SDKs..."
+
+  sdks_directory="$SDK_REPO_DIR/sdks"
+
+  if [[ ! -d $sdks_directory ]]; then
+    echo "[WARN] Directory $sdks_directory doesn't exists, creating it..."
+    mkdir $sdks_directory
+    echo "[INFO] Directory $sdks_directory created."
+  fi
+
+  cp -r "$GENERATOR_REPO_DIR/generated-sources/$LANGUAGE/." "$SDK_REPO_DIR/sdks"
+
+  echo "[INFO] Copy successful."
```

Comment:
Should we echo another message if previous "cp" failed?

Reply:
indeed we should not thanks